### PR TITLE
Move CI away from Conda

### LINKED ISSUE #348: Move CI to not use conda
Cartopy now has wheels available on PyPI (meaning that you can just use pip to install it, no conda needed, see #345).

This means that we also don't need micromamba on the CI, and can instead use pip.

We could also decide to use [uv](https://github.com/astral-sh/uv) to speed up the CI even more. This would just mean replacing any `pip` command with `uv pip`. 
For example, `uv pip install wam2layers[dev]`.

A complete install with `pip install uv; uv pip install wam2layers[complete]` took 36 s on my system (clean venv).
After removing that venv and doing `pip install wam2layers[complete]` in a new venv, it took 54 s.

### PR DESCRIPTION
**Pull request by BSchilperoort**
closes #348

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -21,26 +21,25 @@ jobs:
...
-          wam2layers --help
-        shell: bash -el {0}
-      - name: Run pytest
-        run: pytest --basetemp=tmp/
-        shell: micromamba-shell {0}
+          python3 -m pip install --upgrade pip
+      - name: Install WAM2Layers
+        run: |
+          python3 -m pip install .[dev]
+      - name: Run unit tests
```
Comment (Peter9192) [inline .github/workflows/ci.yaml:40]:
```suggestion
      - name: Run tests
```

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -11,7 +11,7 @@ on:
 jobs:
 
   build:
-    name: Micromamba test on (${{ matrix.python-version }}, ${{ matrix.os }})
+    name: Unit tests on (${{ matrix.python-version }}, ${{ matrix.os }})
```
Comment (Peter9192) [inline .github/workflows/ci.yaml:14]:
```suggestion
    name: Run tests on (${{ matrix.python-version }}, ${{ matrix.os }})
```

**Review by Peter9192 — APPROVED** [review]
Nice to get rid of mamba. Simpler = better. Most important tests are regression tests rather than unit tests :-)

**Comment by Peter9192** [comment]
Shall we add a changelog entry for this as well?

**Comment by BSchilperoort** [comment]
Thanks for the review, @Peter9192 !

**Pull request merged**